### PR TITLE
Docs: update dependencies

### PR DESCRIPTION
**Pull request by humitos**
Documentation build process is failing currently.
This commit should fix it.

### CODE REVIEW

**Comment by humitos** [comment]
Meh, I got tired of trying 🙃

**Review by agjohnson — APPROVED** [review]

**Comment by humitos** [comment]
I wasn't able to make it work, tho 😬

File: setup.cfg [hunk 1]
```diff
@@ -29,9 +29,13 @@ install_requires =
...
-    Jinja2~=2.3
-    MarkupSafe~=2.0.0
-    Sphinx~=4.5.0
-    sphinx_rtd_theme~=1.0.0
+    # We can't upgrade Sphinx because of sphinxcontrib-autoanysrc (Sphinx<5) and sphinx-js (Sphinx<6)
+    Sphinx<5
+    # Newer versions require Sphinx>=5
+    sphinxcontrib-applehelp==1.0.4
+    sphinxcontrib-devhelp==1.0.2
+    sphinxcontrib-htmlhelp==1.0.4
```
Comment (agjohnson) [inline setup.cfg:37]: This is close to my fix in other repos:

```toml
sphinxcontrib-applehelp = "1.0.4"
sphinxcontrib-devhelp = "1.0.2"
sphinxcontrib-htmlhelp = "2.0.1"
sphinxcontrib-qthelp = "1.0.3"
sphinxcontrib-serializinghtml = "1.1.5"
```

Comment (humitos) [reply setup.cfg]: I was able to upgrade Sphinx to 5.x. So, we are not facing this problem anymore here.

Originally, it wasn't possible to upgrade due to `sphinxcontrib-autoanysrc`, but I opened an issue there and the maintainer released a new version 🎉 

https://github.com/sphinx-contrib/autoanysrc/issues/3

**Pull request merged**